File: problems/3000/Solution.scala
object Solution {
    def areaOfMaxDiagonal(dimensions: Array[Array[Int]]): Int = {
        var bestDiag = 0
        var bestArea = 0
        for (d <- dimensions) {
            val length = d(0)
            val width = d(1)
            val diag = length * length + width * width
            val area = length * width
            if (diag > bestDiag || (diag == bestDiag && area > bestArea)) {
                bestDiag = diag
                bestArea = area
            }
        }
        bestArea
    }
}

File: problems/3000/solution.cpp
class Solution {
public:
    int areaOfMaxDiagonal(vector<vector<int>>& dimensions) {
        int bestDiag = 0, bestArea = 0;
        for (const auto& d : dimensions) {
            int length = d[0], width = d[1];
            int diag = length * length + width * width;
            int area = length * width;
            if (diag > bestDiag || (diag == bestDiag && area > bestArea)) {
                bestDiag = diag;
                bestArea = area;
            }
        }
        return bestArea;
    }
};
